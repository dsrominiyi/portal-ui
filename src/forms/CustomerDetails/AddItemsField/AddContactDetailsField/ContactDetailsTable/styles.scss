.contact-details-table {
  margin-top: 1rem;
}
