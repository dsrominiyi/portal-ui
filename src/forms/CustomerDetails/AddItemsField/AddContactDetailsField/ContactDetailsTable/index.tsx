import React, { FunctionComponent } from 'react';

import TableDisplay from '../../../../../components/TableDisplay';

import styles from './styles.scss';

export const ContactDetailsTable: FunctionComponent<ItemTableProps> = ({
  items: contactDetails,
  onEdit,
  onDelete
}: ItemTableProps) => {
  const rows = (contactDetails as Contact[]).map(contact => {
    const { type, detail } = contact;
    return [`${type}: ${detail}`];
  });

  return (
    <TableDisplay
      additionalClasses={[styles.contactDetailsTable]}
      headers={['Contact']}
      rows={rows}
      onEdit={onEdit}
      onDelete={onDelete}
      attributes={{ 'test-id': 'contact-details-table' }}
    />
  );
};

ContactDetailsTable.displayName = 'ContactDetailsTable';
export default ContactDetailsTable;
