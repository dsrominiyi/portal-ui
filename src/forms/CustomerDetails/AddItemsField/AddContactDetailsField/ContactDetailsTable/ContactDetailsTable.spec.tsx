import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import ContactDetailsTable from '.';

let table: ShallowWrapper;
const contactsDetails = [
  { type: ContactType.HomeNumber, detail: '08589392900' },
  { type: ContactType.WorkEmail, detail: 'jsmith@test.com' }
];

describe('ContactDetailsTable', () => {
  beforeEach(() => {
    table = shallow(
      <ContactDetailsTable items={contactsDetails} onEdit={() => null} onDelete={() => null} />
    );
  });

  test('ContactDetailsTable matches snapshot', () => {
    expect(table).toMatchSnapshot();
  });
});
