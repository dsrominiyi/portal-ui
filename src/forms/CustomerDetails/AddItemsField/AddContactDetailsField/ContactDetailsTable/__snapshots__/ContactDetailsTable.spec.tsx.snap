// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactDetailsTable ContactDetailsTable matches snapshot 1`] = `
<TableDisplay
  additionalClasses={
    Array [
      "contactDetailsTable",
    ]
  }
  attributes={
    Object {
      "test-id": "contact-details-table",
    }
  }
  headers={
    Array [
      "Contact",
    ]
  }
  onDelete={[Function]}
  onEdit={[Function]}
  rows={
    Array [
      Array [
        "Home Number: 08589392900",
      ],
      Array [
        "Work Email: jsmith@test.com",
      ],
    ]
  }
/>
`;
